include/fluent-bit/flb_output.h

```diff
@@ -526,6 +565,125 @@ static FLB_INLINE void output_pre_cb_flush(void)
                     persisted_params.config);
 }
 
+/* same as above but for timer coros */
+static FLB_INLINE void output_pre_timer_cb(void)
+{
+    struct flb_coro *coro;
+    struct flb_out_timer_coro_params *params;
+    struct flb_output_instance *o_ins;
+    struct flb_out_thread_instance *th_ins;
+    struct flb_output_timer_coro *timer_coro;
+
+
+    params = (struct flb_out_timer_coro_params *) FLB_TLS_GET(timer_coro_params);
+    if (!params) {
+        flb_error("[output] no timer coro params defined, unexpected");
+        return;
+    }
+
+    coro = params->coro;
+    timer_coro = params->output_timer;
+    o_ins = params->output_timer->o_ins;
+
+    /* Run the callback provided by the output plugin */
+    params->timer_data->cb(params->config, params->timer_data->data);
+
+    /* move coro to destroy queue */
+    if (flb_output_is_threaded(o_ins) == FLB_TRUE) {
+        th_ins = flb_output_thread_instance_get();
+        pthread_mutex_lock(&th_ins->timer_mutex);
+        mk_list_del(&timer_coro->_head);
+        mk_list_add(&timer_coro->_head, &th_ins->timer_coro_list_destroy);
```

Comment:
Todo: understand destroy list logic better

Reply:
Ah yea... I forgot to implement deletion, its in the design: https://github.com/fluent/fluent-bit/issues/7466